lib/mongo_ecto.ex

```diff
@@ -472,28 +472,45 @@ defmodule Mongo.Ecto do
   defp load_objectid(_), do: :error
 
   @doc false
-  def dumpers(:time,            type), do: [type, &dump_time/1]
-  def dumpers(:date,            type), do: [type, &dump_date/1]
-  def dumpers(:utc_datetime,    type), do: [type, &dump_datetime/1]
-  def dumpers(:naive_datetime,  type), do: [type, &dump_datetime/1]
-  def dumpers(:binary_id,       type), do: [type, &dump_objectid/1]
-  def dumpers(:uuid,            type), do: [type, &dump_binary(&1, :uuid)]
-  def dumpers(:binary,          type), do: [type, &dump_binary(&1, :generic)]
-  def dumpers(_base,            type), do: [type]
-
-  defp dump_time({_, _, _, _} = time),
-    do: {:ok, BSON.DateTime.from_datetime({{0, 0, 0}, time})}
+  def dumpers(:time,           type), do: [type, &dump_time/1]
+  def dumpers(:date,           type), do: [type, &dump_date/1]
+  def dumpers(:utc_datetime,   type), do: [type, &dump_utc_datetime/1]
+  def dumpers(:naive_datetime, type), do: [type, &dump_naive_datetime/1]
+  def dumpers(:binary_id,      type), do: [type, &dump_objectid/1]
+  def dumpers(:uuid,           type), do: [type, &dump_binary(&1, :uuid)]
+  def dumpers(:binary,         type), do: [type, &dump_binary(&1, :generic)]
+  def dumpers(_base,           type), do: [type]
+
+  defp dump_time({h, m, s, _} = time),
+    do: Time.from_erl({h, m, s})
   defp dump_time(_),
     do: :error
 
   defp dump_date({_, _, _} = date),
-    do: {:ok, BSON.DateTime.from_datetime({date, {0, 0, 0, 0}})}
+    do: Date.from_erl!(date)
   defp dump_date(_),
     do: :error
 
-  defp dump_datetime({{_, _, _}, {_, _, _, _}} = datetime),
-    do: {:ok, BSON.DateTime.from_datetime(datetime)}
-  defp dump_datetime(_),
+  defp dump_utc_datetime({{_, _, _} = date, {h, m, s, ms}} = v) do
+    datetime =
+    {date, {h, m, s}}
+    |> NaiveDateTime.from_erl!({ms, 6})
```

Comment:
Is this microseconds or milliseconds? Make sure that the sub-second precision is correct for whatever you are using.